### PR TITLE
Deprecate PendingStateManager local state interfaces

### PR DESCRIPTION
**Pull request by kian-thompson**
[How contribute to this repo](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md).

[Guidelines for Pull Requests](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

## Description

All of these interfaces are internal, so we should not export them anymore (makes changes in this area harder).

## Upcoming Breaking Changes

The following interfaces used by the `PendingStateManager` have been deprecated and will no longer be exported in a future version:
- `IPendingMessage`
- `IPendingFlush`
- `IPendingState`
- `IPendingLocalState`

### CODE REVIEW

File: api-report/container-runtime.api.md [hunk 1]
```diff
@@ -430,7 +430,7 @@ export interface IPendingLocalState {
 
 // @public
 export interface IPendingMessage {
```
Comment (anthony-murphy) [inline api-report/container-runtime.api.md:432]: does this even need to be exported? could we consider deprecating the whole class, and just not exporting it anymore?

Comment (kian-thompson) [reply api-report/container-runtime.api.md:432]: @wes-carlson Is there a plan to make `IPendingRuntimeState` exported by runtime? Looks like it's currently casting from an `unknown` type

Comment (wes-carlson) [reply api-report/container-runtime.api.md:432]: No, `IPendingRuntimeState` is internal to runtime and shouldn't be exported.

File: packages/runtime/container-runtime/src/opLifecycle/outbox.ts [hunk 2]
```diff
@@ -113,9 +112,9 @@ export class Outbox {
 
 	private flushInternal(rawBatch: IBatch) {
 		const processedBatch = this.compressBatch(rawBatch);
-		const clientSequenceNumber = this.sendBatch(processedBatch);
+		this.sendBatch(processedBatch);
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/opLifecycle/outbox.ts:115]: I feel like it should be a separate PR.

**Review by vladsud — APPROVED** [review]
Looks good, but could you please separate deprecation and changes around clientSequenceNumber?
